src/status.rs

```diff
@@ -1,1601 +1,331 @@
+            NotFound => Some("Not Found"),
+            MethodNotAllowed => Some("Method Not Allowed"),
+            NotAcceptable => Some("Not Acceptable"),
+            ProxyAuthenticationRequired => Some("Proxy Authentication Required"),
+            RequestTimeout => Some("Request Timeout"),
+            Conflict => Some("Conflict"),
+            Gone => Some("Gone"),
+            LengthRequired => Some("Length Required"),
+            PreconditionFailed => Some("Precondition Failed"),
+            PayloadTooLarge => Some("Payload Too Large"),
+            UriTooLong => Some("URI Too Long"),
+            UnsupportedMediaType => Some("Unsupported Media Type"),
+            RangeNotSatisfiable => Some("Range Not Satisfiable"),
+            ExpectationFailed => Some("Expectation Failed"),
+            ImATeapot => Some("I'm a teapot"),
+
+            UnprocessableEntity => Some("Unprocessable Entity"),
+            Locked => Some("Locked"),
+            FailedDependency => Some("Failed Dependency"),
+
+            UpgradeRequired => Some("Upgrade Required"),
+
+            PreconditionRequired => Some("Precondition Required"),
+            TooManyRequests => Some("Too Many Requests"),
+
+            RequestHeaderFieldsTooLarge => Some("Request Header Fields Too Large"),
+
+            InternalServerError => Some("Internal Server Error"),
+            NotImplemented => Some("Not Implemented"),
+            BadGateway => Some("Bad Gateway"),
+            ServiceUnavailable => Some("Service Unavailable"),
+            GatewayTimeout => Some("Gateway Timeout"),
+            HttpVersionNotSupported => Some("HTTP Version Not Supported"),
+            VariantAlsoNegotiates => Some("Variant Also Negotiates"),
+            InsufficientStorage => Some("Insufficient Storage"),
+            LoopDetected => Some("Loop Detected"),
+
+            NotExtended => Some("Not Extended"),
+            NetworkAuthenticationRequired => Some("Network Authentication Required"),
+            _ => None
         }
     }
 
     /// Determine the class of a status code, based on its first digit.
     pub fn class(&self) -> StatusClass {
-        let code = *self as u16;  // Range of possible values: 100..599.
-        // We could match 100..199 &c., but this way we avoid unreachable!() at the end.
-        if code < 200 {
-            StatusClass::Informational
+        let code = self.to_u16().unwrap();
+        // TODO rewrite
+        if code < 100 {
+            NoClass
+        } else if code < 200 {
+            Informational
         } else if code < 300 {
-            StatusClass::Success
+            Successful
         } else if code < 400 {
-            StatusClass::Redirection
+            Redirection
         } else if code < 500 {
-            StatusClass::ClientError
+            ClientError
+        } else if code < 600{
+            ServerError
         } else {
-            StatusClass::ServerError
+            NoClass
         }
     }
-
```

Comment:
Why remove all these convenience methods? The allow you figure out the class without requiring an additional import.